src/main/java/sirius/kernel/cache/ManagedCacheRemoverBuilder.java

```diff
@@ -0,0 +1,86 @@
+package sirius.kernel.cache;
+
+import sirius.kernel.commons.Tuple;
+
+import java.util.function.BiFunction;
+import java.util.function.BiPredicate;
+
+class ManagedCacheRemoverBuilder<K, V, T> implements CacheRemoverBuilder<K, V, T> {
+    private final Cache<K, V> cache;
+    private final String discriminator;
+
+    /**
+     * The combined mapping/filter function up to now.
+     * The {@link State} defines whether the object was already marked for removal via {@link #removeAlways} or
+     * marked as filtered via {@link #filter}. If it is marked as undecided, it should be processed further.
+     */
+    private final BiFunction<String, CacheEntry<K, V>, Tuple<T, State>> mapper;
+
+    private enum State {
```

Comment:
the impl. is IMHO fine as it is, just as an additional thought: couldn't we have also provided two special instances of subclasses which either always or never remove and return these as "dummies" (which then would ignore all successive calls to map / filter / remove) (?)